### PR TITLE
Adds a new steal objective, the Supermatter Sliver. [WIP]

### PR DESCRIPTION
**Pull request by harmonyn**
:cl: bandit
add: The Syndicate has been observed with supermatter technology. Nanotrasen is investigating potential tampering with its supermatter devices.
add: Traitors can steal supermatter slivers now. It works much like the plutonium core.
/:cl:

Traitors have a new steal objective, to extract a sliver of a supermatter crystal. This works much like stealing the plutonium core:

- Use a special scalpel (different fluff is welcome) to slice a sliver off. Like the plutonium core, this sliver will periodically emit radiation if not stored.
- Pick the sliver up with tongs, also provided. If you or anything else touches the crystal, then it dusts. If you use the tongs to attack a person, the resulting force will dust you as well (so you can't go around instakilling people). The sliver will also be destroyed, to prevent a chain reaction of people dusting people.
- Use the tongs to put it in the container, provided as well.
- Once you extract a sliver, the supermatter will gain energy, requiring you to have protection. The exact amount of energy can be tweaked, the idea is that delamination will not happen immediately but that it is serious, noticeable sabotage and the traitor should probably get out fast.
- Both the scalpel and tongs do burn instead of brute damage, because they are made of SEARING COLD FREON THIS IS GREAT FLUFF sigh

What this adds to the game: Steal objectives should be more difficult than they are. No one gives a shit about, say, the blueprints. The supermatter also lacks a meaningful traitor interaction that is not just releasing a singularity. This provides that. 

This is a work in progress and there are still bugs, but I wanted to get the idea and basic functionality down.

**TO DO:**
- Finish the tongs-dusting interactions, including properly getting rid of the sliver if you dust something with it (it is microscopic, after all)
- Prevent people from dragging the sliver
- Prevent people from throwing the tongs + sliver like a javelin
- Prevent infinite slivers from being made, turning the SM into a radiation deathtrap (or do I want this?)
- You can only extract a sliver of an active supermatter crystal? possibly?
- Enabling people to open the container, and by extension the nuke core container?
- Make this all work with shards from cargo too, if tators want to do that

### CODE REVIEW

**Comment by PKPenguin321** [comment]
>The sliver will also be destroyed, to prevent a chain reaction of people dusting people.

Lame!

**Comment by Tokiko1** [comment]
Consider making the sliver not dust and disappear, but rather burn off a random bodypart from both people but stay around.

File: code/game/objects/items/theft_tools.dm [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	if(!iscarbon(user))
+		return FALSE
+	var/mob/ded = user
+	to_chat(user, "<span class='warning'>You reach for the supermatter sliver. That was dumb.</span>")
+	radiation_pulse(get_turf(user), 2, 4, 50, 1)
+	ded.dust()
+
+/obj/item/nuke_core_container/supermatter
+	name = "supermatter receptacle"
+	desc = "A tiny receptacle that releases an inert freon mix upon sealing, allowing a sliver of a supermatter crystal to be safely stored.."
```
Comment (swindly) [inline code/game/objects/items/theft_tools.dm]: ..

**Comment by Bawhoppen** [comment]
lore: eswords are powered by supermatter slivers

**Comment by RandomMarine** [comment]
We already have tons of traitors bungling at stealing the nuke core and dying from radiation.

If this gets merged I'll look forward to watching all the tots dust themselves.

**Comment by Armhulen** [comment]
Make it a supermatter shard instead of sliver

**Comment by bgobandit** [comment]
we already have supermatter shards and they are not items

**Comment by octareenroon91** [comment]
Got something against slivers?
Maybe you'd like splinters or dust or flakes?

Consider syndie whisk-broom and dust pan for stealing supermatter dust.

**Comment by Armhulen** [comment]
It's a giant crystal, be creative

**Comment by fludd12** [comment]
"The syndicate cannot bypass the Nanotrasen engine core monopoly. Harvest and deliver the Supermatter crystalline growth seed by extracting it from an active Supermatter."

File: code/game/objects/items/theft_tools.dm [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+		STOP_PROCESSING(SSobj, sliver)
+		icon_state = "core_container_sealed"
+		playsound(loc, 'sound/items/Deconstruct.ogg', 60, 1)
+		if(ismob(loc))
+			to_chat(loc, "<span class='warning'>[src] is permanently sealed, [sliver] is safely contained.</span>")
+
+/obj/item/nuke_core_container/supermatter/attackby(obj/item/weapon/hemostat/supermatter/tongs, mob/user)
+	if(istype(tongs))
+		if(!user.temporarilyRemoveItemFromInventory(tongs))
+			to_chat(user, "<span class='warning'>The [tongs] is stuck to your hand!</span>")
```
Comment (kevinz000) [inline code/game/objects/items/theft_tools.dm]: How will the tongs being stuck to your hand prevent you from operating it while in your hand to release the sliver..?

Comment (bgobandit) [inline code/game/objects/items/theft_tools.dm]: sanity check I guess

**Comment by cebutris** [comment]
The special tongs should be a special hemostat, just to keep the surgery vibe going. Also, they should replace the special supermatter scalpel/hemostat (or tongs) should come in the syndicate surgery dufflebag, like how the nuke screwdriver comes in the fully loaded toolbox

**Comment by bgobandit** [comment]
they're not really a hemostat, the sprite is just similar (actually I think I even used the retractor sprite)

**Comment by cebutris** [comment]
Well yeah, I'm just suggesting that it be a hemostat (or a retractor, I guess) instead of tongs. I was thinking is if one step is a surgery tool, why not have another step use another surgery tool? The hemostat works well for this, because it's good at picking up small things.

**Comment by bgobandit** [comment]
but... they are as much a hemostat as the scalpel is??

File: code/game/objects/items/theft_tools.dm (hunk 2 above)
Comment (Tacolizard) [inline code/game/objects/items/theft_tools.dm]: edward tongerhands

**Comment by Tacolizard** [comment]
the tongs should be called supermatter salad tongs

**Comment by Tacolizard** [comment]
you know tong is a really weird word like is that even a real word or is it slang

**Comment by cebutris** [comment]
New recipe: Supermatter salad. Looks and is crafted like a regular salad, but with supermatter sliver. Dusts anyone who eats it.

**Comment by Tacolizard** [comment]
it must be made with the supermatter tongs though

**Comment by Tacolizard** [comment]
@WJohn come make a salad tongs sprite we need it for the lore

**Comment by cebutris** [comment]
All chefs now get salad tongs to make salads. Syndicate cooking set contains supermatter salad tongs.

**Comment by Tacolizard** [comment]
im gonna crash this pr with no survivors

salad tongs
salad tongs
salad tongs
salad tongs

**Comment by Anonmare** [comment]
Gonna be difficult to harvest anything from the SM if competent engies are in charge (>inb4 "Competent" engies)
Still doable though

**Comment by ExcessiveUseOfCobblestone** [comment]
> Supermatter tongs should be makeshift hemostats

fund it.

**Comment by swindly** [comment]
Engineers shouldn't get this objective.

**Comment by bgobandit** [comment]
why not? they have the same problems everyone else does (other engies being around). we don't prevent HoPs from getting steal objectives or the CMO from the hypospray, etc

**Comment by swindly** [comment]
CMOs are excluded from stealing the hypospray in line 54 of objective_items.dm.

**Comment by optimumtact** [comment]
bog bandit opening a work in progress pr so they can sit on the tracker for five weeks taking forever to respond.

**Comment by bgobandit** [comment]
it's a work day you fruit

**Comment by YakumoChen** [comment]
going to be the devil's advocate and disapprove any objective that encourages delaminating the engine. It's hard enough for the engineers to keep maintained, much less fix it, and we've just started to improve from it blowing up literally every round only recenrly

**Comment by YakumoChen** [comment]
Also keep in mind that an active supermatter will have all the problems of space wind and a supercooled chamber, requiring gear checks much more focused and difficult then stealing the nuke core.

File: code/game/objects/items/theft_tools.dm [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+		radiation_pulse(get_turf(src), 1, 4, 40, 1)
+
+//nuke core box, for carrying the core
+/obj/item/nuke_core_container
+	name = "nuke core container"
+	desc = "Solid container for radioactive objects."
+	icon = 'icons/obj/nuke_tools.dmi'
+	icon_state = "core_container_empty"
+	item_state = "tile"
+	var/obj/item/nuke_core/core = null
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: No `= null`

File: code/game/objects/items/theft_tools.dm [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	icon_state = "core_container_loaded"
+	to_chat(user, "<span class='warning'>Container is sealing...</span>")
+	addtimer(CALLBACK(src, .proc/seal), 50)
+	return 1
+
+/obj/item/nuke_core_container/proc/seal()
+	if(istype(core))
+		STOP_PROCESSING(SSobj, core)
+		icon_state = "core_container_sealed"
+		playsound(loc, 'sound/items/Deconstruct.ogg', 60, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: playsound(src, ...

File: code/game/objects/items/theft_tools.dm [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+/obj/item/nuke_core_container/proc/load(obj/item/nuke_core/ncore, mob/user)
+	if(core || !istype(ncore))
+		return 0
+	ncore.forceMove(src)
+	core = ncore
+	icon_state = "core_container_loaded"
+	to_chat(user, "<span class='warning'>Container is sealing...</span>")
+	addtimer(CALLBACK(src, .proc/seal), 50)
+	return 1
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: TRUE

File: code/game/objects/items/theft_tools.dm [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+	name = "nuke core container"
+	desc = "Solid container for radioactive objects."
+	icon = 'icons/obj/nuke_tools.dmi'
+	icon_state = "core_container_empty"
+	item_state = "tile"
+	var/obj/item/nuke_core/core = null
+
+/obj/item/nuke_core_container/proc/load(obj/item/nuke_core/ncore, mob/user)
+	if(core || !istype(ncore))
+		return 0
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: FALSE

File: code/game/objects/items/theft_tools.dm [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+	item_state = "supermattersliver"
+	pulseicon = "supermatter_sliver_pulse"
+
+/obj/item/nuke_core/supermatter_sliver/attackby(obj/item/W, mob/living/user, params)
+	if(istype(W, /obj/item/weapon/hemostat/supermatter))
+		var/obj/item/weapon/hemostat/supermatter/tongs = W
+		if (tongs.sliver)
+			to_chat(user, "<span class='notice'>\The [tongs] is already holding a supermatter sliver!</span>")
+			return
+		src.forceMove(tongs)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: remove src

File: code/game/objects/items/theft_tools.dm [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+		return FALSE
+	var/mob/ded = user
+	to_chat(user, "<span class='warning'>You reach for the supermatter sliver. That was dumb.</span>")
+	radiation_pulse(get_turf(user), 2, 4, 50, 1)
+	ded.dust()
+
+/obj/item/nuke_core_container/supermatter
+	name = "supermatter receptacle"
+	desc = "A tiny receptacle that releases an inert freon mix upon sealing, allowing a sliver of a supermatter crystal to be safely stored.."
+	var/obj/item/nuke_core/supermatter_sliver/sliver = null
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: remove the `= null`

Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: You also need to QDEL_NULL this in `Destroy()`

File: code/game/objects/items/theft_tools.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: `\The`

File: code/game/objects/items/theft_tools.dm [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+		icon_state = "core_container_sealed"
+		playsound(loc, 'sound/items/Deconstruct.ogg', 60, 1)
+		if(ismob(loc))
+			to_chat(loc, "<span class='warning'>[src] is permanently sealed, [sliver] is safely contained.</span>")
+
+/obj/item/nuke_core_container/supermatter/attackby(obj/item/weapon/hemostat/supermatter/tongs, mob/user)
+	if(istype(tongs))
+		if(!user.temporarilyRemoveItemFromInventory(tongs))
+			to_chat(user, "<span class='warning'>The [tongs] is stuck to your hand!</span>")
+			return
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: This is a trailing return, remove it

File: code/game/objects/items/theft_tools.dm [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+	else
+		investigate_log("has consumed [AM].", "supermatter")
+		qdel(AM)
+	user.visible_message("<span class='danger'>As [user] touches \the [AM] with \a [src], silence fills the room...</span>",\
+			"<span class='userdanger'>You touch \the [AM] with \the [src], and everything suddenly goes silent.</span>\n<span class='notice'>\The [AM] flashes into dust, and soon as you can register this, you do as well.</span>",\
+			"<span class='italics'>Everything suddenly goes silent.</span>")
+	radiation_pulse(get_turf(user), 2, 4, 50, 1)
+	user.dust()
+	icon_state = "supermatter_tongs"
+	qdel(sliver)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: QDEL_NULL

File: code/game/objects/items/theft_tools.dm [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+	usesound = 'sound/weapons/bladeslice.ogg'
+
+/obj/item/weapon/hemostat/supermatter
+	name = "supermatter extraction tongs"
+	desc = "A pair of tongs made from condensed freon gas, searingly cold to the touch, that can safely grip a supermatter sliver."
+	icon = 'icons/obj/nuke_tools.dmi'
+	icon_state = "supermatter_tongs"
+	toolspeed = 0.75
+	damtype = "fire"
+	var/obj/item/nuke_core/supermatter_sliver/sliver = null
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: remove the `= null`

Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: QDEL_NULL it in `Destroy()`

File: code/game/objects/items/theft_tools.dm [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+/obj/item/nuke_core_container/supermatter/load(obj/item/weapon/hemostat/supermatter/T, mob/user)
+	if(!istype(T) || !T.sliver)
+		return 0
+	T.sliver.forceMove(src)
+	sliver = T.sliver
+	T.sliver = null
+	icon_state = "core_container_loaded"
+	to_chat(user, "<span class='warning'>Container is sealing...</span>")
+	addtimer(CALLBACK(src, .proc/seal), 50)
+	return 1
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: TRUE

File: code/game/objects/items/theft_tools.dm [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+	ded.dust()
+
+/obj/item/nuke_core_container/supermatter
+	name = "supermatter receptacle"
+	desc = "A tiny receptacle that releases an inert freon mix upon sealing, allowing a sliver of a supermatter crystal to be safely stored.."
+	var/obj/item/nuke_core/supermatter_sliver/sliver = null
+
+/obj/item/nuke_core_container/supermatter/load(obj/item/weapon/hemostat/supermatter/T, mob/user)
+	if(!istype(T) || !T.sliver)
+		return 0
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: FALSE

File: code/game/objects/items/theft_tools.dm [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+		qdel(W)
+
+/obj/item/nuke_core/supermatter_sliver/pickup(mob/living/user)
+	..()
+	if(!iscarbon(user))
+		return FALSE
+	var/mob/ded = user
+	to_chat(user, "<span class='warning'>You reach for the supermatter sliver. That was dumb.</span>")
+	radiation_pulse(get_turf(user), 2, 4, 50, 1)
+	ded.dust()
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: play the supermatter consume sound

File: code/game/objects/items/theft_tools.dm [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+	damtype = "fire"
+	var/obj/item/nuke_core/supermatter_sliver/sliver = null
+
+/obj/item/weapon/hemostat/supermatter/afterattack(atom/O, mob/user, proximity)
+	if(!sliver)
+		return
+	if(ismovableatom(O))
+		Consume(O)
+
+/obj/item/weapon/hemostat/supermatter/proc/Consume(atom/movable/AM, mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/theft_tools.dm]: play the supermatter consume sound

File: code/modules/power/supermatter/supermatter.dm [hunk 16]
```diff
@@ -461,7 +461,14 @@
...
 		return
-	if(user.drop_item(W))
+	if(istype(W, /obj/item/weapon/scalpel/supermatter))
+		playsound(loc, W.usesound, 100, 1)
+		to_chat(user, "<span class='notice'>You carefully begin to scrape [src] with [W]...</span>")
+		if(do_after(user, 60*W.toolspeed,target=src))
+			to_chat(user, "<span class='notice'>You extract a sliver from [src]. [src] begins to react violently!</span>")
+			new /obj/item/nuke_core/supermatter_sliver(user.loc)
+			matter_power += 200
+	else if(user.drop_item(W))
```
Comment (Cyberboss) [inline code/modules/power/supermatter/supermatter.dm]: user.dropItemToGround(W)

File: code/modules/power/supermatter/supermatter.dm [hunk 17]
```diff
@@ -461,7 +461,14 @@
 /obj/machinery/power/supermatter_shard/attackby(obj/item/W, mob/living/user, params)
 	if(!istype(W) || (W.flags & ABSTRACT) || !istype(user))
 		return
-	if(user.drop_item(W))
+	if(istype(W, /obj/item/weapon/scalpel/supermatter))
+		playsound(loc, W.usesound, 100, 1)
```
Comment (Cyberboss) [inline code/modules/power/supermatter/supermatter.dm]: playsound(src, ...

File: code/modules/power/supermatter/supermatter.dm [hunk 18]
```diff
@@ -461,7 +461,14 @@
...
+		to_chat(user, "<span class='notice'>You carefully begin to scrape [src] with [W]...</span>")
+		if(do_after(user, 60*W.toolspeed,target=src))
+			to_chat(user, "<span class='notice'>You extract a sliver from [src]. [src] begins to react violently!</span>")
```
Comment (Cyberboss) [inline code/modules/power/supermatter/supermatter.dm]: The text entries should probably be prefixed with \the

File: tgstation.dme [hunk 19]
```diff
@@ -173,7 +173,6 @@
 #include "code\controllers\subsystem\inbounds.dm"
 #include "code\controllers\subsystem\ipintel.dm"
 #include "code\controllers\subsystem\job.dm"
-#include "code\controllers\subsystem\language.dm"
```
Comment (Cyberboss) [inline tgstation.dme]: What?

File: tgstation.dme [hunk 20]
```diff
@@ -970,7 +969,6 @@
 #include "code\modules\admin\admin_investigate.dm"
 #include "code\modules\admin\admin_ranks.dm"
 #include "code\modules\admin\admin_verbs.dm"
-#include "code\modules\admin\adminmenu.dm"
```
Comment (Cyberboss) [inline tgstation.dme]: What?

File: tgstation.dme [hunk 21]
```diff
@@ -1671,7 +1669,6 @@
 #include "code\modules\mob\living\simple_animal\hostile\hivebot.dm"
 #include "code\modules\mob\living\simple_animal\hostile\hostile.dm"
 #include "code\modules\mob\living\simple_animal\hostile\illusion.dm"
-#include "code\modules\mob\living\simple_animal\hostile\jungle_mobs.dm"
```
Comment (Cyberboss) [inline tgstation.dme]: What?

File: tgstation.dme [hunk 22]
```diff
@@ -1965,7 +1962,6 @@
 #include "code\modules\shuttle\arrivals.dm"
 #include "code\modules\shuttle\assault_pod.dm"
 #include "code\modules\shuttle\computer.dm"
-#include "code\modules\shuttle\elevator.dm"
```
Comment (Cyberboss) [inline tgstation.dme]: What?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: tgstation.dme (hunk 19 above)
Comment (bgobandit) [inline tgstation.dme]: I don't know how any of these came about. I pulled to master and based this off the new master branch, and I had to re-check a bunch of shit. wtf do I do, commit the DME?

File: code/modules/power/supermatter/supermatter.dm (hunk 18 above)
Comment (bgobandit) [inline code/modules/power/supermatter/supermatter.dm]: oops

**Comment by Anonmare** [comment]
Does this have TK checks?

**Comment by bgobandit** [comment]
not yet, that said, I don't know how to handle TK because, like, if you telekinetically grab supermatter, that isn't the same thing as coming into physical contact

**Comment by Cyberboss** [comment]
We added a gib for tk sm touches recently

On May 12, 2017 10:16 PM, "bgobandit" <notifications@github.com> wrote:

not yet, that said, I don't know how to handle TK because, like, if you
telekinetically grab supermatter, that isn't the same thing as coming into
physical contact

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/27137#issuecomment-301219281>,
or mute the thread

**Comment by bgobandit** [comment]
@Cyberboss can you address my comment about the includes please it is kind of critical for testing

File: tgstation.dme (hunk 19 above)
Comment (optimumtact) [inline tgstation.dme]: pull the DME from our latest master and replace yours

**Comment by Bawhoppen** [comment]
make it so you have to use the special surgery tools to extract the core of the SM which beats like a heart BECAUSE IT'S ALIVE

**Comment by fludd12** [comment]
Actually the SM being a living organism with a pulsing crystal heart sounds awesome and metal as frick

**Comment by bgobandit** [comment]
if someone wants to sprite that feel free

**Comment by bgobandit** [comment]
btw this is not stale i'm just having problems testing it because my dream daemon is not registering my client, causing hella runtimes.

File: tgstation.dme (hunk 19 above)
Comment (Tacolizard) [inline tgstation.dme]: secret plot to remove lizard language confirmed?

File: tgstation.dme (hunk 20 above)
Comment (Tacolizard) [inline tgstation.dme]: why?

File: tgstation.dme (hunk 21 above)
Comment (Tacolizard) [inline tgstation.dme]: how?

File: tgstation.dme (hunk 22 above)
Comment (Tacolizard) [inline tgstation.dme]: when?

**Comment by Jalleo** [comment]
@bgobandit @Cyberboss maybe able to help.

**Comment by optimumtact** [comment]
That's not lore friendly

**Comment by AdamElTablawy** [comment]
It doesn't dust people when thrown, right?

**Comment by octareenroon91** [comment]
>It doesn't dust people when thrown, right?

That might be OK if it destroys itself and costs your greentext?

**Comment by lzimann** [comment]
@bgobandit

**Comment by bgobandit** [comment]
I've said like 50000 times I cannot test this because my client is runtiming and no one has suggested a cause or a solution

**Comment by Jalleo** [comment]
I think its the cleanbot bug. Try updating your Repo and turn on testing. If that does not solve it talk to @Cyberboss or @MrStonedOne. Preferably on IRC.

**Comment by Cyberboss** [comment]
Wouldn't be the cleanbot bug if he's not running linux

**Comment by MrStonedOne** [comment]
I was able to connect just fine when i launched a test server.

Hmmmmmmmmm

**Comment by lzimann** [comment]
Ping whenever you keep going with it.

**Pull request closed without merging**